static/app/components/dropdownMenuControlV2.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  ElementType,
+  HTMLAttributes,
+  MouseEvent,
+  ReactNode,
+  RefObject,
+  useRef,
+} from 'react';
+import styled from '@emotion/styled';
+import {useButton} from '@react-aria/button';
+import {AriaMenuOptions, useMenuTrigger} from '@react-aria/menu';
+import {AriaPositionProps, OverlayProps} from '@react-aria/overlays';
+import {Item, Section} from '@react-stately/collections';
+import {useMenuTriggerState} from '@react-stately/menu';
+import {MenuTriggerProps} from '@react-types/menu';
+
+import DropdownButton, {DropdownButtonProps} from 'sentry/components/dropdownButtonV2';
+import {MenuItemProps} from 'sentry/components/dropdownMenuItemV2';
+import Menu from 'sentry/components/dropdownMenuV2';
+
+type TriggerProps = {
+  props: HTMLAttributes<HTMLButtonElement> & {
+    onClick?: (e: MouseEvent) => void;
+  };
+  ref: RefObject<HTMLButtonElement>;
+};
+
+type Props = {
+  /**
+   * Items to display inside the dropdown menu. If the item has
+   * a `children` prop, it will be rendered as a menu section.
+   * If it has a `children` prop and its `isSubmenu` prop
+   * is true, it will be rendered as a submenu.
```

Comment:
minor nit, but what are you using to wrap your comments? Just eyeballing it?

Reply:
oh woah just discovered that my commenting package has shortcut for reformatting paragraphs 🤯